chain-abstraction/interfaces.go

```diff
@@ -254,20 +256,23 @@ func (e EdgeStatus) String() string {
 }
 
 type OriginHeights struct {
-	BlockChallengeOriginHeight   Height `json:"blockChallengeOriginHeight"`
-	BigStepChallengeOriginHeight Height `json:"bigStepChallengeOriginHeight"`
+	ChallengeOriginHeights []Height `json:"challengeOriginHeights"`
 }
 
-type ChallengeLevel uint8
-
 // ReadOnlyEdge defines methods that only retrieve data from the chain
 // regarding for a given edge.
 type ReadOnlyEdge interface {
 	// The unique identifier for an edge.
 	Id() EdgeId
-	// The type of challenge the edge is a part of.
-	GetType() EdgeType
+	// The challenge level the edge is a part of.
 	GetChallengeLevel() (ChallengeLevel, error)
+	// GetReversedChallengeLevel obtains the challenge level for the edge. The lowest level starts at 0, and goes all way
```

Comment:
This is a bit janky, but haven't found a good workaround at this time. It's really helpful for us to have challenge levels start at small step being 0 in some places, but 2 in others